paddle/fluid/operators/mkldnn/concat_mkldnn_op.cc

```diff
@@ -134,7 +182,65 @@ class ConcatMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
     auto concat = prim_creator.CreateConcatPrimitive(concat_pd, output, place);
     stream(stream::kind::eager).submit({concat}).wait();
 
-    output->set_mkldnn_prim_desc(concat_pd.dst_primitive_desc());
+    output->set_layout(DataLayout::kMKLDNN);
+    output->set_format(GetDstMemFormat(concat_pd));
```

Comment:
It is advised to stick to the new `set_mkldnn_prim_desc` API created by @jczaja, since the old `set_format` and `set_layout` are going to be deprecated.

Reply:
Thanks for your suggestion, i have refine it.

Reply:
> since the old set_format and set_layout are going to be deprecated

@Sand3r- @jczaja Do you need to use `set_mkldnn_prim_desc` to replace all the `set_format` and `set_layout` in the codes?

Reply:
That's the plan, yes.